src/export.rs

```diff
@@ -73,7 +73,11 @@ pub fn export(pin_config: &PinConfig, symlink_root: Option<&str>) -> Result<(),
         try!(fs::create_dir_all(symroot));
 
         // set the pin direction
-        try!(pin_config.get_pin().set_direction(pin_config.direction.clone()));
+		    // Call up to 3 times, waiting 100 milliseconds after each unsuccessful attempt.
+        match retry(3, 100, || pin_config.get_pin().set_direction(pin_config.direction.clone()), |res| res.is_ok()) {
```

Comment:
A couple things on this line:

- Please pull out the constants with names like `PIN_CONFIGURE_RETRIES` and `PIN_CONFIGURE_TIMEOUT`
- This should probably be split across multiple lines, it's getting a bit long. [`rustfmt`](https://crates.io/crates/rustfmt) will do this automatically for you.